app/controllers/StudentController.php

```diff
@@ -391,6 +481,8 @@ function confirm() {
 
           // Params to vars
           extract($data = $this->data($params));
+          extract($this->registrationData($params));
```

Comment:
Since you are only extracting one var ($registration), its prob better to just have that be clear:

$registration = self::registrationData($params);